styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/TransitionLogger.java

```diff
@@ -23,26 +23,37 @@
 import static java.lang.String.format;
 
 import com.google.common.annotations.VisibleForTesting;
-import com.spotify.styx.state.OutputHandler;
+import com.spotify.styx.model.SequenceEvent;
 import com.spotify.styx.state.RunState;
+import com.spotify.styx.util.EventUtil;
+import java.util.Locale;
 import java.util.Objects;
+import java.util.function.BiConsumer;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class TransitionLogger implements OutputHandler {
+public class TransitionLogger implements BiConsumer<SequenceEvent, RunState> {
 
-  private static final Logger LOG = LoggerFactory.getLogger(TransitionLogger.class);
+  private final Logger log;
 
-  private final String prefix;
+  public TransitionLogger() {
+    this(LoggerFactory.getLogger(TransitionLogger.class));
+  }
 
-  public TransitionLogger(String prefix) {
-    this.prefix = Objects.requireNonNull(prefix);
+  TransitionLogger(Logger log) {
+    this.log = Objects.requireNonNull(log, "log");
   }
 
   @Override
-  public void transitionInto(RunState state) {
-    final String name = state.state().name().toLowerCase();
-    LOG.info("{}{} transition -> {} {}", prefix, state.workflowInstance(), name, stateInfo(state));
+  public void accept(SequenceEvent sequenceEvent, RunState state) {
+    var stateName = state.state().name().toLowerCase(Locale.ROOT);
+    log.info("{} transition #{} {}({}) -> {} {}",
```

Comment:
This will break `user/processed-internal-event` logs metric in styx GCP project. We can fix later.

Reply:
😫 
Scraping unstructured logs for metrics is not a great practice.

Reply:
We don't event seem to be using this metric. It's not in any dashboard I can find. If we care about it, let's make it a real metric.